subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/listener/TestMetricsSender.kt

```diff
@@ -0,0 +1,66 @@
+package com.avito.runner.service.listener
+
+import com.avito.android.stats.GaugeMetric
+import com.avito.android.stats.StatsDSender
+import com.avito.logger.LoggerFactory
+import com.avito.logger.create
+import com.avito.runner.service.model.TestCase
+import com.avito.runner.service.model.TestCaseRun
+import com.avito.runner.service.worker.device.Device
+import com.avito.time.TimeProvider
+
+class TestMetricsSender(
+    private val statsDSender: StatsDSender,
+    private val timeProvider: TimeProvider,
+    loggerFactory: LoggerFactory
+) : TestListener {
+
+    private val prefix = "runner.test.duration"
+
+    private val logger = loggerFactory.create<TestMetricsSender>()
+
+    private val intendedMarks = mutableMapOf<Key, Long>()
+
+    private val startedMarks = mutableMapOf<Key, Long>()
+
+    override fun intended(test: TestCase, targetPackage: String, executionNumber: Int) {
+        intendedMarks[Key(test, executionNumber)] = timeProvider.nowInMillis()
+    }
+
+    override fun started(device: Device, targetPackage: String, test: TestCase, executionNumber: Int) {
+        val key = Key(test, executionNumber)
+        startedMarks[key] = timeProvider.nowInMillis()
+
+        val intendedMark = intendedMarks[key]
+
+        if (intendedMark != null) {
+            statsDSender.send(prefix, GaugeMetric("queue", timeProvider.nowInMillis() - intendedMark))
```

Comment:
I didn't get what do we want to collect here.
It sends the same gauge metric `runner.test.duration.queue` on each execution. 
If I'm not mistaken, only the last value within a bucket (30 sec by default) will be preserved.
https://statsd.readthedocs.io/en/v3.2.1/types.html#gauges

Reply:
yep, i added test coordinates to the metric

Reply:
![Screenshot from 2021-01-15 20-21-36](https://user-images.githubusercontent.com/1104540/104758521-9f44ba80-576f-11eb-9735-a8c5e4323434.png)

We've discussed the main idea.
Detalization is a subject of discussion.